lib/translator/translation.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common.h"
+
+namespace translator {
+
+StatusCode Translation::ScsiToNvme(absl::Span<const uint8_t> scsi_cmd,
```

Comment:
Would it be more idiomatic for the functionality of this method to take place in the class constructor, or is this fine?
Currently, interaction with the library expects a strict adherence to ScsiToNvme->GetNvmeCmds->NvmeToScsi. Is this fine from an idiomatic standpoint @Ralith ?

Reply:
Constructors cannot fail except by throwing an exception, which isn't usually appropriate in general and is specifically forbidden by the Google style guide. An alternative would be a static method which returns a new object or an error, but because we don't have convenient access to templates that make that easy, the approach you currently have is much simpler.

Additionally, downstream code might want to keep a preallocated table of these structures and rotate through them, in which case repeatedly iterating that cycle on a long-lived object is convenient.